src/main/java/com/williamfiset/algorithms/search/BinarySearch.java

```diff
@@ -26,7 +26,11 @@ public static double binarySearch(
     do {
 
       // Find the middle point
-      mid = (hi + lo) / 2.0;
+      //mid = (hi + lo) / 2.0;  // Here is the bug, Cause when dealing with big or huge number, overflow will be happened 
+      
+      //The following line to fix it 
+      mid = lo + ((hi - lo) / 2.0)
```

Comment:
I think it's cleaner and easier to read if you just use the unsigned shift as `mid = (hi + lo) >>> 1` to handle the overflow. This is how Java implements their built in binary search. And please remove the comments

https://stackoverflow.com/questions/19058859/what-does-mean-in-java

Reply:
Okay. This is so good

Reply:
I've added.